### PR TITLE
feat: add possibility to request some stats about loaded CAD sectors

### PR DESCRIPTION
**Pull request by larsmoa**
Makes it easier to understand how much data is loaded at any given time.

### CODE REVIEW

File: examples/src/utils/OverrideSectorCuller.ts [hunk 1]
```diff
@@ -28,9 +28,19 @@ export class OverrideSectorCuller implements reveal.internal.SectorCuller {
...
+  ) {
     if (this._wantedSectors) {
-      return this._wantedSectors;
+      const spendage: reveal.internal.CadModelSectorLoadStatistics = {
+        downloadSize : 0,
+        drawCalls : 0,
+        loadedSectorCount : 0,
+        simpleSectorCount : 0,
+        detailedSectorCount : 0,
+        forcedSectorCount : 0,
```
Comment (christjt) [inline examples/src/utils/OverrideSectorCuller.ts:39]: Should probably rename this to something more descriptive. `forcedDetailSectorCount` f.ex

File: viewer/src/datamodels/cad/CadModelUpdateHandler.ts [hunk 2]
```diff
@@ -70,12 +81,15 @@ export class CadModelUpdateHandler {
       ]).pipe(map(makeClippingInput)),
       this.loadingModelObservable()
     ]);
+    const spendageReportCb = (spendage: SectorLoadingSpendage) => {
```
Comment (christjt) [inline viewer/src/datamodels/cad/CadModelUpdateHandler.ts:84]: not familiar with the Cb suffix, please rename

File: viewer/src/datamodels/cad/sector/rxSectorUtilities.ts [hunk 3]
```diff
@@ -14,15 +14,18 @@ import { filter, switchMap, tap, publish, subscribeOn } from 'rxjs/operators';
...
 ): OperatorFunction<DetermineSectorsInput, ConsumedSector> {
   return publish((source$: Observable<DetermineSectorsInput>) => {
     const updateSector = (input: DetermineSectorsInput) => {
       const { cameraInMotion } = input;
       if (cameraInMotion) {
         return EMPTY;
       }
-      return from(sectorCuller.determineSectors(input)).pipe(
+      const prioritizedResult = sectorCuller.determineSectors(input);
+      spendageReportCb(prioritizedResult.spendage);
```
Comment (christjt) [inline viewer/src/datamodels/cad/sector/rxSectorUtilities.ts:27]: Again, don't understand Cb, does this generate a report, add data to a report?

**Review by christjt — CHANGES_REQUESTED** [review]
Nice! Overall looks good, just a few variable names that need to be a little more descriptive in my opinion.

File: viewer/src/datamodels/cad/sector/rxSectorUtilities.ts (hunk 3 above)
Comment (larsmoa) [reply viewer/src/datamodels/cad/sector/rxSectorUtilities.ts]: Tbh, the callback is horrible. I just got tired of fighting RxJS to make a "function" that returns a list of nodes + some statistics in some sensible way.

Renamed the callback.

**Review by christjt — APPROVED** [review]

**Pull request merged**